functions/go/set-namespace/main.go

```diff
@@ -33,11 +38,31 @@ func main() {
 					yaml.FieldSetter{StringValue: namespace}); err != nil {
 					return err
 				}
+			} else {
+				if !allowClusterScoped {
+					apiVersionNode, err := node.Pipe(yaml.Lookup("apiVersion"))
+					if err != nil {
+						continue
+					}
+					apiVersion := yaml.GetValue(apiVersionNode)
+					nameNode, err := node.Pipe(yaml.Lookup("metadata", "name"))
+					if err != nil {
+						continue
+					}
+					name := yaml.GetValue(nameNode)
+					resID := fmt.Sprintf("apiVersion: %s, kind: %s, name: %s", apiVersion, kind, name)
+					clusterScopedResources = append(clusterScopedResources, resID)
+				}
 			}
 		}
+		if len(clusterScopedResources) > 0 {
+			return fmt.Errorf("the app config should only include namespace-scoped resources. "+
+				"But the following cluster-scoped resources are found:\n%s\n", strings.Join(clusterScopedResources, "\n"))
+		}
 		return nil
 	})
-	cmd.Flags().StringVar(&namespace, "namespace", "", "the namespace value")
+	cmd.Flags().StringVar(&namespace, "namespace", "", "the namespace to be added into namespaced resources")
```

Comment:
move flags to the top

Reply:
These flags can only be added after `cmd` is defined.